build.gradle

```diff
@@ -122,6 +122,7 @@ subprojects {
     tasks.withType(JavaCompile).configureEach {
         options.compilerArgs += [
             '-Xlint:all,-processing',
+            '-Xlint:all,-deprecation',
```

Comment:
I share the concern. It is impractical when needing to actually find errors. The balance between 'nice-to-have hopefully we'll get to it and meanwhile we ignore this output' vs 'this is getting in the way and takes up real time when finding errors which *must* be fixed', the latter needs to win out.

Since the concern is noise in output, we could tune down the number of warnings, perhaps instead of 1000, do 10.

Reply:
I removed the deprecation suppression for now so we can move forward. I'll submit a follow-up PR, I'm thinking reducing the warnings might be the better way to go.

Otherwise, I tested out PMD violations, adding/removing them, they are still detected (and I'll note, *much* faster than before)